first draft with images (#243)
shipshapecode/website-nuxt: content/blog/posts/2022-07-01-accessibility-basics.md

To ensure everyone is able to use your website, it is important to make sure the underlying structure of each page is set up in a way that works with the different devices and tools people may be using to interact with them.

| ![AOL landing page from 1999, comprised of mostly low-resolution graphic ads and blue underlined links to different parts of the web](/img/blog/AOL-1999.png) |
| :-----------------------------------------------------------------------------------------------------------------------------------------------------------: |
|                                                                    _Remember this beauty?_                                                                    |

<br><br/>

## Navigating a Modern Website with Accessibility Tools

| ![shipshape.io landing page, newly updated this past year](/img/blog/shipshape-home.png) |
| :--------------------------------------------------------------------------------------: |
|                  _How do you process the content on this landing page?_                  |

<br/>

> A visual flow that may instantly make sense for those of us with normal vision and the ability to navigate the GUI with a mouse or trackpad can be vastly different navigational experience for someone without those abilities.

Reviewer: ```suggestion
A visual flow that may instantly make sense for those of us with normal vision and the ability to navigate the GUI with a mouse or trackpad can be a vastly different experience for someone without those abilities.
```

Some folks may rely on tools like a screenreader or keyboard to be able to access the digital world. These tools rely on software engineers to create apps with proper DOM structure so their users can to get to the information they need quickly and efficiently.

## Generate the DOM Structure First

<br/>

![shipshape.io landing page marked up with arrows starting at the header and zig-zagging across the page to each text area](/img/blog/shipshape-home-arrows.png)

If you are not visually impaired, chances are likely that your eyes jumped to scan each header on the page, starting at the top left. You might have next read a section of body copy, if the header caught your interest. Though your focus may have been on the text, your eyes concurrently processed the two images of our mascot at work "building" apps, which are opposite each segment of copy. You could likely skim through the buttons made up of tech logos on the bottom right-hand corner of the page.

> Someone without those same abilities could have a fairly similar experience if the page is built with their experience in mind. Based on our abilities as able-bodied developers, our instinct may be to add elements as we see them on a design spec.

Reviewer: ```suggestion
Someone without those same abilities could have a fairly similar experience if the page is built with their experience in mind. Based on our abilities as developers without a visual disability, our instinct may be to add elements as we see them on a design spec.
```

Reviewer: Because the article content is specific to people with a visual disability I checked [this guide](https://www.acedisability.org.au/information-for-providers/language-disability.php) for preferred language. Might be useful where you see fit!

> When building with accessibility in mind, a good rule of thumb is to start by thinking of what elements should be on the page _without_ any of the design in place. Ccreate the scaffolding for the content first, then add in the creative touches that differentiate your page.

Reviewer 2: ```suggestion
When building with accessibility in mind, a good rule of thumb is to start by thinking of what elements should be on the page _without_ any of the design in place. Create the scaffolding for the content first, then add in the creative touches that differentiate your page.
```

```html
<section class="flex">
  <div class="order-last">
    <h1>Title</h1>
    <p>some body copy</p>
  </div>
  <div>
    <img />
  </div>
</section>
```

### Use Proper H-Tags

Accessibility tools rely on well-organized DOM and content to direct their users to where they want to get on a page. Screen reader users are given the option to listen to a list of headings on a page so they can skip around and get to their desired section for further interaction _without_ going through every single element on the page, line by line.

An anti-pattern for using `h-tag`'s is to set each header level to a certain set of styles and sizes and using them for different text styles through the page. This promotes [using headers out of order](https://www.w3.org/WAI/tutorials/page-structure/headings/). Instead, first make sure your content is organized in proper order, then create the CSS styles to meet your design needs.

> ![Section of shipshape.io with blocks of text using 5 different styles: large-font blue header, large-font white header, small, bolded white text within a button, small regular-weight blue body text, and small regular-weight white body text](/img/blog/h-tag-don't.png)

Reviewer: ```suggestion
<img src="/img/blog/h-tag-don't.png" align="left" style="padding: 2rem"></img>
```

Reviewer: Hope this is closer to what you described but feel free to play around with it!

Author: Thank you, yes!

Lighthouse scans through your web app and offers a rundown of what is and isn't considered accessible. It can also measure your app's quality for load speed and search engine optimization. Keep in mind scoring 100% does _not_ necessarily mean your site is perfectly accessible. Manual tests are recommended to fix any gaps the tool may have missed.

### Color Contrast Checkers

Lighhouse will let you know about sections of your website that pass/fail color contrast. You can quickly determine at what colors and sizes of text have enough contrast with their background color by using contrast checkers like [WCAG Color contrast checker](https://chrome.google.com/webstore/detail/wcag-color-contrast-check/plnahcmalebffmaghcpcmpaciebdhgdf?hl=en) or [Colour Contrast Checker](https://chrome.google.com/webstore/detail/colour-contrast-checker/nmmjeclfkgjdomacpcflgdkgpphpmnfe?hl=en-GB).

> Your primary text should have enough contrast to be readable by folks with different types of colorblindness. Additional text states like `hover` or `active` don’t necessarily need to pass the contrast test as long as you have other cues that someone is over a link for example (i.e.: by using semantic html 😉). 

Reviewer 2: ```suggestion
Your primary text should have enough contrast to be readable by folks with different types of color blindness. Additional text states like `hover` or `active` don’t necessarily need to pass the contrast test as long as you have other cues that someone is over a link for example (i.e.: by using semantic html 😉). 
```

---
authorId: hannakim
categories:

>   - '2022'

Reviewer 3: ```suggestion
```

>   - web accessibility

Reviewer 3: ```suggestion
  - a11y
```
